test/test_torch.py

```diff
@@ -12586,36 +12566,37 @@ def test_scatter_add_non_unique_index(self, device):
                          torch.tensor([[3], [1]], device=device,
                                       dtype=torch.float32).repeat(1, width))
 
-    @onlyCPU
     def test_scatter_reduce_non_unique_index(self, device):
         height = 2
         width = 2
         index = torch.zeros(height, width, dtype=torch.long, device=device)
-        test_data = [
-            (torch.ones(height, width, device=device, dtype=torch.float32),
-             torch.ones(height, width, device=device, dtype=torch.float32),
-             torch.tensor([[3], [1]], device=device, dtype=torch.float32).repeat(1, width), "add"),
-
-            (torch.ones(height, width, device=device, dtype=torch.float32),
-             torch.ones(height, width, device=device, dtype=torch.float32),
-             torch.tensor([[-1], [1]], device=device,
-                          dtype=torch.float32).repeat(1, width), "subtract"),
-
-            (torch.tensor([2], device=device, dtype=torch.float32).repeat(height, width),
-             torch.tensor([2], device=device, dtype=torch.float32).repeat(height, width),
-             torch.tensor([[8], [2]], device=device,
-                          dtype=torch.float32).repeat(1, width), "multiply"),
-
-            (torch.tensor([2], device=device, dtype=torch.float32).repeat(height, width),
-             torch.tensor([2], device=device, dtype=torch.float32).repeat(height, width),
-             torch.tensor([[0.5], [2]], device=device,
-                          dtype=torch.float32).repeat(1, width), "divide"),
-        ]
-
-        for input, src, result, operation in test_data:
-            input.scatter_(0, index, src, reduce=operation)
-            self.assertEqual(input, result)
+        for dtype in torch.testing.get_all_fp_dtypes(
+                include_bfloat16=device.startswith('cuda')) + torch.testing.get_all_complex_dtypes():
+            test_data = [
+                (torch.ones(height, width, device=device, dtype=dtype),
+                 torch.ones(height, width, device=device, dtype=dtype),
+                 torch.tensor([[3], [1]], device=device, dtype=dtype).repeat(1, width), "add"),
+                (torch.tensor([2], device=device, dtype=dtype).repeat(height, width),
+                 torch.tensor([2], device=device, dtype=dtype).repeat(height, width),
+                 torch.tensor([[8], [2]], device=device,
+                              dtype=dtype).repeat(1, width), "multiply"),
+            ]
 
+            for input, src, result, operation in test_data:
+                if operation == "multiply" and torch.is_complex(input):
+                    continue
+                input.scatter_(0, index, src, reduce=operation)
+                self.assertEqual(input, result, msg=f"result: {result} input: {input} method: {str(operation)}")
+    @onlyCUDA
+    def test_scatter_reduce_multiply_unsupported_dtypes(self, device):
```

Comment:
Add is supported for all types. I think simply adding the decorators should be enough since we have the same test for add and multiply with just an if condition that skips integer/complex dtype testing for multiply.